test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAppsWithFrameworkRefs.cs

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using FluentAssertions;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Xml.Linq;
+using Xunit;
+using static Microsoft.NET.TestFramework.Commands.MSBuildTest;
+
+namespace Microsoft.NET.Build.Tests
+{
+    public class GivenThatWeWantToBuildAppsWithFrameworkRefs
+    {
+        private TestAssetsManager _testAssetsManager = TestAssetsManager.TestProjectsAssetsManager;
+
+        [Fact]
+        public void It_builds_the_projects_successfully()
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return;
+            }
+
+            var testAsset = _testAssetsManager
+                .CopyTestAsset("AppsWithFrameworkReferences")
+                .WithSource();
+
+            testAsset.Restore("EntityFrameworkApp");
+            testAsset.Restore("StopwatchLib");
+
+            VerifyProjectsBuild(testAsset);
+        }
+
+        [Fact]
+        public void It_builds_with_disable_implicit_frameworkRefs()
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return;
+            }
+
+            var testAsset = _testAssetsManager
+                .CopyTestAsset("AppsWithFrameworkReferences")
+                .WithSource()
+                .WithProjectChanges(project =>
+                {
+                    // Some framework references may be provided implicitly.
+                    // Turn off this implicit addition to verify that references are
+                    // provided in this scenario by project assets file
+
+                    var ns = project.Root.Name.Namespace;
+                    var propertyGroup = project.Root.Elements(ns + "PropertyGroup").FirstOrDefault();
+                    propertyGroup.Should().NotBeNull();
+
+                    propertyGroup.Add(new XElement(ns + "DisableImplicitFrameworkReferences", "true"));
```

Comment:
You were probably copying this from code that I wrote, but it might just be easier to invoke the build command with a `/p:DisableImplicitFrameworkReferences=true` parameter, rather than modifying the XML here.